packages/block-library/src/navigation-menu/index.php

```diff
@@ -15,20 +15,69 @@
  * @return string Returns the post content with the legacy widget added.
  */
 function render_block_navigation_menu( $attributes, $content, $block ) {
-	return '<nav class="wp-block-navigation-menu">' . build_navigation_menu_html( $block ) . '</nav>';
+	// Add CSS classes and inline styles.
+	$bg_color_css_classes = '';
+	if ( array_key_exists('backgroundColor', $attributes ) ) {
+		$bg_color_css_classes .= ' has-background-color';
+	}
+	if ( array_key_exists('backgroundColorCSSClass', $attributes ) ) {
+		$bg_color_css_classes .= " {$attributes['backgroundColorCSSClass']}";
+	}
+	$bg_color_css_classes = trim( $bg_color_css_classes );
+
+	$text_color_css_classes = '';
+	if ( array_key_exists('textColor', $attributes ) ) {
+		$text_color_css_classes .= ' has-text-color';
+	}
+
+	if ( array_key_exists('textColorCSSClass', $attributes ) ) {
+		$text_color_css_classes .= " ${attributes['textColorCSSClass']}";
+	}
+	$text_color_css_classes = trim( $text_color_css_classes );
+
+
+	$bg_inline_styles = '';
+	if ( array_key_exists('customBackgroundColor', $attributes ) ) {
+		$bg_inline_styles = "background-color: ${attributes['customBackgroundColor']};";
+	} elseif ( array_key_exists('backgroundColorValue', $attributes ) ) {
+		$bg_inline_styles = "background-color: ${attributes['backgroundColorValue']};";
+	}
+
+	$text_inline_styles = '';
+	if ( array_key_exists('textColorValue', $attributes ) ) {
+		$text_inline_styles = " color: ${attributes['textColorValue']};";
+	} elseif ( array_key_exists('customTextColor', $attributes ) ) {
+		$text_inline_styles = " color: ${attributes['customTextColor']};";
+	}
+
+	return
+		'<nav class="wp-block-navigation-menu">' .
+			build_navigation_menu_html(
+				$block,
+				$bg_color_css_classes,
+				$text_color_css_classes,
+				$bg_inline_styles,
+				$text_inline_styles
```

Comment:
Rather than parsing it all out here and passing it to `build_navigation_menu_html()`, I'd rather do it in `build_navigation_menu_html()` directly. All necessary information seems to be in `$block['attrs']`.

Reply:
The data to build the CSS classes and the inline styles actually comes from the `$attributes` parameter and, beyond we could pass this parameter straigh to the `build_navigation_menu_html()` function, the attribute belongs to the `<NavigationMenu />`, not to the `<NavigationMenuItem />`.
Let me twaek this passing an colors array instead.